src/hist/plot.py

```diff
@@ -214,185 +262,408 @@ def plot2d_full(
+    def gauss(
+        x: np.ndarray,
+        constant: float = constant,
+        mean: float = mean,
+        sigma: float = sigma,
+    ) -> np.ndarray:
+        # Note: Force np.ndarray type as numpy ufuncs have type "Any"
+        ret: np.ndarray = constant * np.exp(
+            -np.square(x - mean) / (2 * np.square(sigma))
+        )
+        return ret
+
+    return gauss
+
+
+def _fit_callable_to_hist(
+    model: Callable[[np.ndarray], np.ndarray],
+    histogram: hist.BaseHist,
+    likelihood: bool = False,
+) -> "Tuple[np.ndarray, np.ndarray, np.ndarray, Tuple[Tuple[float, ...], np.ndarray]]":
+    """
+    Fit a model, a callable function, to the histogram values.
+    """
+    variances = histogram.variances()
+    if variances is None:
+        raise RuntimeError(
+            "Cannot compute from a variance-less histogram, try a Weight storage"
+        )
+    hist_uncert = np.sqrt(variances)
+
+    # Infer best fit model parameters and covariance matrix
+    xdata = histogram.axes[0].centers
+    popt, pcov = _curve_fit_wrapper(
+        model, xdata, histogram.values(), hist_uncert, likelihood=likelihood
+    )
+    model_values = model(xdata, *popt)
+
+    if np.isfinite(pcov).all():
+        n_samples = 100
+        vopts = np.random.multivariate_normal(popt, pcov, n_samples)
+        sampled_ydata = np.vstack([model(xdata, *vopt).T for vopt in vopts])
+        model_uncert = np.nanstd(sampled_ydata, axis=0)
+    else:
+        model_uncert = np.zeros_like(hist_uncert)
+
+    return model_values, model_uncert, hist_uncert, (popt, pcov)
+
+
+def _plot_fit_result(
+    __hist: hist.BaseHist,
+    model_values: np.ndarray,
+    model_uncert: np.ndarray,
+    ax: matplotlib.axes.Axes,
+    eb_kwargs: Dict[str, Any],
+    fp_kwargs: Dict[str, Any],
+    ub_kwargs: Dict[str, Any],
+) -> FitResultArtists:
+    """
+    Plot fit of model to histogram data
+    """
+    x_values = __hist.axes[0].centers
+    variances = __hist.variances()
+    if variances is None:
+        raise RuntimeError(
+            "Cannot compute from a variance-less histogram, try a Weight storage"
+        )
+    hist_uncert = np.sqrt(variances)
+
+    errorbars = ax.errorbar(x_values, __hist.values(), hist_uncert, **eb_kwargs)
+
+    # Ensure zorder draws data points above model
+    line_zorder = errorbars[0].get_zorder() - 1
+    (line,) = ax.plot(x_values, model_values, **fp_kwargs, zorder=line_zorder)
+
+    # Uncertainty band for fitted function
+    ub_kwargs.setdefault("color", line.get_color())
+    if ub_kwargs["color"] == line.get_color():
+        ub_kwargs.setdefault("alpha", 0.3)
+    uncertainty_band = ax.fill_between(
+        x_values,
+        model_values - model_uncert,
+        model_values + model_uncert,
+        **ub_kwargs,
+    )
+
+    return FitResultArtists(line, errorbars, uncertainty_band)
+
+
+def plot_ratio(
+    __hist: hist.BaseHist,
+    ratio: np.ndarray,
```

Comment:
> FYI, this is the changes I tried; no fixes but added strings / callables and removed quotes in plot.py (since matplotlib is available there) (in .patch form if you want to apply them):

Thanks for the patch as an example. :+1: I had already gotten there when I saw this, but they are very similar with the only exception being that in `plot.py` `plot_ratio` and `plot_pull` should use `np.ndarray` and not `Union[str, Callable[[np.ndarray], np.ndarray], np.ndarray]` as at this point these are both just plotting the already calculated ratios or pulls onto the subplot axis.